tests/unit/utils/test_app_utils.py

```diff
@@ -79,235 +78,162 @@ async def test_application_upgrade_packages_with_hold(model):
-    model, current_release_value, target_release_value
+    mock_get_current_osd_release,
+    mock_get_required_osd_release,
+    model,
+    current_required_osd_release,
+    current_osd_release,
 ):
-    check_result = f"""
-        crush_version 7
-        require_min_compat_client luminous
-        min_compat_client jewel
-        require_osd_release {current_release_value}
-        foo bar test
-    """
-    model.run_on_unit.side_effect = [
-        {"Code": "0", "Stdout": check_result},
-        {"Code": "0", "Stdout": "Success"},
-    ]
+    mock_get_required_osd_release.return_value = current_required_osd_release
+    mock_get_current_osd_release.return_value = current_osd_release
 
-    await app_utils.set_require_osd_release_option(
-        unit="ceph-mon/0", model=model, ceph_release=target_release_value
-    )
+    await app_utils.set_require_osd_release_option(unit="ceph-mon/0", model=model)
 
-    model.run_on_unit.assert_called_once_with(
-        unit_name="ceph-mon/0",
-        command="ceph osd dump",
-        timeout=600,
-    )
+    assert not model.run_on_unit.called
 
 
 @pytest.mark.asyncio
-async def test_set_require_osd_release_option_check_unsuccessful(model):
-    exp_error_msg = (
-        "Cannot determine the current value of require_osd_release on ceph-mon unit 'ceph-mon/0'."
+async def test_get_required_osd_release(model):
+    expected_current_release = "octopus"
+    check_result = """
+        {"crush_version":7,"min_compat_client":"jewel","require_osd_release":"octopus"}
+    """
+    model.run_on_unit.return_value = {"Code": "0", "Stdout": check_result}
+    actual_current_release = await app_utils._get_required_osd_release(
+        unit="ceph-mon/0", model=model
     )
-    model.run_on_unit.return_value = {"Code": "non-zero", "Stderr": "error"}
-
-    with pytest.raises(RunUpgradeError, match=exp_error_msg):
-        await app_utils.set_require_osd_release_option(
-            unit="ceph-mon/0", model=model, ceph_release="pacific"
-        )
 
     model.run_on_unit.assert_called_once_with(
         unit_name="ceph-mon/0",
-        command="ceph osd dump",
+        command="ceph osd dump -f json",
         timeout=600,
     )
+    assert actual_current_release == expected_current_release
 
 
 @pytest.mark.asyncio
-async def test_set_require_osd_release_option_check_error(model):
-    exp_error_msg = (
-        "Cannot determine the current value of require_osd_release on ceph-mon unit 'ceph-mon/0'."
+async def test_get_current_osd_release(model):
+    expected_osd_release = "octopus"
+    check_output = """
+    {
+        "mon": {
+            "ceph version 15.2.17 (8a82819d84cf884bd39c17e3236e0632) octopus (stable)": 1
+        },
+        "mgr": {
+            "ceph version 15.2.17 (8a82819d84cf884bd39c17e3236e0632) octopus (stable)": 1
+        },
+        "osd": {
+            "ceph version 15.2.17 (8a82819d84cf884bd39c17e3236e0632) %s (stable)": 3
```

Comment:
If we are not parameterizing, why not use `octopus` directly here?

Reply:
Because I think it it is clearer to see what we are expecting with `expected_osd_release`.